auslib/web/admin/views/validators.py

```diff
@@ -0,0 +1,147 @@
+        return True
+    # empty input is fine
+    if field_value is None or field_value == '':
+        return True
+    try:
+        op, operand = get_op(field_value)
+        logger.debug('Got (%s, %s) from get_op', op, operand)
+    except TypeError:
+        # get_op field returns None if no operator or no match, can't be unpacked
+        raise JsonschemaValidationError("Invalid input for buildID : %s. No Operator or Match found." % field_value)
+    return True
+
+
+@draft4_format_checker.checks(format="version", raises=JsonschemaValidationError)
+def version_validator(field_value):
+    logger.debug('starting in version_validator: version data is %s' % field_value)
+    if not isinstance(field_value, str_types):
+        return True
+    # empty input is fine
+    if field_value is None or field_value == '':
+        return True
+    rules_version_list = field_value.split(",")
+    is_list_of_versions = len(rules_version_list) > 1
+    for rule_version in rules_version_list:
+        try:
+            op, operand = get_op(rule_version)
+            if is_list_of_versions and op != operator.eq:
+                raise JsonschemaValidationError('Invalid input for %s .Relational Operators are not allowed'
+                                                ' when providing a list of versions.' % field_value)
+            version = MozillaVersion(operand)
+        except JsonschemaValidationError:
+            raise
+        except ValueError:
+            raise JsonschemaValidationError("ValueError. Couldn't parse version for %s. Invalid '%s' input value"
+                                            % (field_value, field_value))
+        except:
+            raise JsonschemaValidationError('Invalid input for %s . No Operator or Match found.' % field_value)
+        # MozillaVersion doesn't error on empty strings
+        if not hasattr(version, 'version'):
+            raise JsonschemaValidationError("Couldn't parse the version for %s. No attribute 'version' was detected."
+                                            % field_value)
+    return True
+
+
+@draft4_format_checker.checks(format="priority", raises=JsonschemaValidationError)
+def priority_validator(field_value):
+    logger.debug('starting in priority_validator: field data is %s' % field_value)
+    if not isinstance(field_value, str_types) and not isinstance(field_value, int) and field_value is not None:
+        return False
+    # empty input is fine
+    if field_value is None or field_value == '':
+        return True
+    try:
+        x = int(field_value)
+    except:
+        raise JsonschemaValidationError('Invalid input %s for priority.Not an integer.' % field_value)
+    if x < 0:
+        raise JsonschemaValidationError("Priority field value %s shouldn't be less than 0" % field_value)
+    return True
+
+
+@draft4_format_checker.checks(format="backgroundRate", raises=JsonschemaValidationError)
+def background_rate_validator(field_value):
```

Comment:
Can you refactor this to minimize duplication between priority_validator and background_rate_validator? It seems like you could support both of them by writing a function that allows a minimum and maximum value to be specified.